File: src/main/java/me/pljr/pljrapi/objects/PLJRSound.java
package me.pljr.pljrapi.objects;

import org.bukkit.Bukkit;
import org.bukkit.Location;
import org.bukkit.Sound;
import org.bukkit.entity.Player;

public class PLJRSound {
    private final Sound type;
    private final float volume;
    private final float pitch;

    public PLJRSound(Sound type, float volume, float pitch){
        this.type = type;
        this.volume = volume;
        this.pitch = pitch;
    }

    public Sound getType() {
        return type;
    }

    public float getVolume() {
        return volume;
    }

    public float getPitch() {
        return pitch;
    }

    public void play(Player player){
        Location playerLoc = player.getLocation();
        player.playSound(playerLoc, type, volume, pitch);
    }

    public void broadcast(){
        for (Player player : Bukkit.getOnlinePlayers()){
            play(player);
        }
    }
}

File: src/main/kotlin/me/pljr/pljrapispigot/sound/PLJRSound.kt
package me.pljr.pljrapispigot.sound

import org.bukkit.Bukkit
import org.bukkit.Sound
import org.bukkit.entity.Player

class PLJRSound(val type: Sound, val volume: Float, val pitch: Float) {

    fun play(player: Player) = player.playSound(player.location, type, volume, pitch)

    fun broadcast() = Bukkit.getOnlinePlayers().forEach { play(it) }
}